From 17103b653e515c9fd5edb24dcd45ccd1af6022b6 Mon Sep 17 00:00:00 2001
From: Soumen Kumar Dey <sdey@nvidia.com>
Date: Mon, 24 Sep 2012 19:24:11 -0700
Subject: [PATCH 5/6] libstagefright: Allow other H264 profiles for encoder

Remove hack that always sets it to baseline. Stagefright should
listen to codec capabilities

Change-Id: I5d4aa881274802a2c8a9b1501248c1a422c1ced9
Reviewed-on: http://git-master/r/134963
(cherry picked from commit 51868bbc30ad99f13b79698bc60dc0ba017d615f)
---
 media/libstagefright/OMXCodec.cpp |    7 -------
 1 files changed, 0 insertions(+), 7 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index b96b4ae..c4eec69 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -1196,13 +1196,6 @@ status_t OMXCodec::setupAVCEncoderParameters(const sp<MetaData>& meta) {
     h264type.eProfile = static_cast<OMX_VIDEO_AVCPROFILETYPE>(profileLevel.mProfile);
     h264type.eLevel = static_cast<OMX_VIDEO_AVCLEVELTYPE>(profileLevel.mLevel);
 
-    // XXX
-    if (h264type.eProfile != OMX_VIDEO_AVCProfileBaseline) {
-        ALOGW("Use baseline profile instead of %d for AVC recording",
-            h264type.eProfile);
-        h264type.eProfile = OMX_VIDEO_AVCProfileBaseline;
-    }
-
     if (h264type.eProfile == OMX_VIDEO_AVCProfileBaseline) {
         h264type.nSliceHeaderSpacing = 0;
         h264type.bUseHadamard = OMX_TRUE;
-- 
1.7.1

